plugins/catalog-backend/src/catalog/DatabaseLocationsCatalog.ts

```diff
@@ -16,11 +16,24 @@
 
 import { Database } from '../database';
 import { AddLocation, Location, LocationsCatalog } from './types';
+import { LocationReaders } from '../ingestion';
 
 export class DatabaseLocationsCatalog implements LocationsCatalog {
   constructor(private readonly database: Database) {}
 
   async addLocation(location: AddLocation): Promise<Location> {
+    const outputs = await LocationReaders.create().read(
+      location.type,
+      location.target,
+    );
+    outputs.forEach(output => {
+      if (output.type === 'error') {
+        throw new Error(
+          `Can't read location at ${location.target} with error: ${output.error.message}`,
+        );
+      }
+    });
```

Comment:
at this point, i intended the entities to be stored as well - essentially, the end behavior would be as if a refresh loop had run and finished, when this call returns. To get that to work, maybe you will want to refactor slightly in DatabaseManager or so

Reply:
Sound interesting. I'll get to it.

Reply:
We decided to not do this as part of this pr